### PR TITLE
Update for 21b feature - Improve performance ~4x

### PR DESCRIPTION
**Pull request by jhughes-mw**

### CODE REVIEW

**Comment by rdmello** [comment]
Looks good to me!
I think some of the copyrights still say "2014" and probably need to be updated to 2023.

**Comment by jhughes-mw** [comment]
> Looks good to me! I think some of the copyrights still say "2014" and probably need to be updated to 2023.

Thanks, I only updated the dates if I moved the copyright line or modified the file. The others are all whitespace only changes, so I didn't update those.

File: +each/+iterators/AllPermutationIterator.m [hunk 1]
```diff
@@ -10,17 +10,17 @@
     %
     % See Also: each, eachPermutation, each.iterators.Iterable
     %
-    
-    %   Copyright 2014 The MathWorks, Inc.
-    
+
+    % Copyright 2014 The MathWorks, Inc.
```
Comment (rdmello) [inline +each/+iterators/AllPermutationIterator.m:14]: Copyrights are out of date

**Review by rdmello — APPROVED** [review]

File: +each/+iterators/AllPermutationIterator.m (hunk 1 above)
Comment (jhughes-mw) [reply +each/+iterators/AllPermutationIterator.m:14]: I didn't update them for whitespace only changes

**Pull request merged**